iup/eval_projections.py

```diff
@@ -1,5 +1,100 @@
+import polars as pl
 import altair as alt
 
+# get mspe #
+def get_mspe(
+        data_df,
+        pred_df,
+        var
+):
+    """
+    Calculate the mean squared prediction error between the observed
+    data and prediction when both are available.
+
+    Parameters:
+    --------------
+    data_df: polars.DataFrame
+        The observed vaccine uptake. Must include `date` and the variable-
+        of-interest (var).
+    pred_df: polars.DataFrame
+        The predicted daily uptake. Must include `date` and the variable-
+        of-interest (var).
+    var: string
+        The name of the variable-of-interest.
+
+    Return:
+    -------------
+
+    A polars.DataFrame with the MSPE and the date of initializing forecast.
+
+    """
+
+    if not any(data_df['date'].is_in(pred_df['date'])):
+        ValueError('No matched dates between data_df and pred_df.')
+
+    common_dates = data_df.filter(
+        pl.col('date').is_in(pred_df['date'])
+    ).select('date')
+
+    if len(common_dates)!= common_dates.n_unique():
+        ValueError('Duplicated dates are found in data_df or pred_df.')
+
+
+    mspe = data_df.join(
+            pred_df,
+            on = 'date',
+            how = 'inner'
+        ).with_columns(
+            spe = (pl.col(var) - pl.col(f"{var}_right"))**2
+        ).with_columns(
+            mspe = pl.col('spe').mean(),
+        ).filter(
+            pl.col('date') == pl.col('date').min(),
+        ).select(
+            'date','mspe'
+        )
```

Comment:
Similarly, these steps are performed by the `get_mspe` method, above. So we should call that method here, rather than explicitly repeating it as a new function.